src/test/java/org/example/Operations/CollectionOperationsTest.java

```diff
@@ -1,250 +1,486 @@
+
+            // Show the metadata no longer exists on the collection.
+            // Same query as before
+            Response queryRes2 = alice.queryOperations().execute_genquery(aliceToken, query, null);
+            assertEquals("Executing genquery request failed", 200, queryRes2.getHttpStatusCode());
+            assertEquals("Executing genquery failed", 0,
+                    getIrodsResponseStatusCode(queryRes2.getBody()));
+            rootNode = assertDoesNotThrow(() -> mapper.readTree(queryRes2.getBody()),
+                    "JsonProcessingException was thrown");
+            int rowsLength = rootNode.path("rows").size();
+            assertEquals(0, rowsLength);
+        } finally {
+            // Remove metadata
+            List<ModifyMetadataOperations> jsonParam2 = new ArrayList<>();
+            jsonParam2.add(new ModifyMetadataOperations("remove", "a1", "v1", "u1"));
+            alice.collections().modify_metadata(aliceToken, collection, jsonParam2, OptionalInt.empty());
+        }
+
+
+    }
+
+    @Test
+    public void testModifyingPermissionsAtomically() {
+        Response res;
+        String collection = "/tempZone/home/alice";
+
+        try {
+            // Give the rodsadmin read permission on the rodsuser's home collection.
+            List<ModifyPermissionsOperations> operation = new ArrayList<>();
+            operation.add(new ModifyPermissionsOperations("rods", "read"));
+            res = assertDoesNotThrow(() ->
+                            alice.collections().modify_permissions(aliceToken, collection, operation, OptionalInt.empty()),
+                    "JsonProcessingException was thrown"
+            );
+
+            assertEquals("Modifying permissions on collection request failed", 200, res.getHttpStatusCode());
+            assertEquals("Modifying permissions on collection failed", 0,
+                    getIrodsResponseStatusCode(res.getBody()));
+
+            // Show the rodsadmin now has permission to read the collection
+            Response statRes = alice.collections().stat(aliceToken, collection, Optional.empty());
+            assertEquals("Stat on collection request failed", 200, statRes.getHttpStatusCode());
+            assertEquals("Stat on collection failed", 0,
+                    getIrodsResponseStatusCode(statRes.getBody()));
+
+            String name = null;
+            String zone = null;
+            String type = null;
+            String perm = null;
+            JsonNode rootNode = assertDoesNotThrow(() -> mapper.readTree(statRes.getBody()),
+                    "JsonProcessingException was thrown");
+            JsonNode permissionsNode = rootNode.path("permissions");
+            Iterator<JsonNode> elements = permissionsNode.elements();
+            while (elements.hasNext()) {
+                JsonNode element = elements.next();
+                name = element.path("name").asText();
+                if (name.equals("rods")) {
```

Comment:
Prefer calling the `.equals` on the string literal to avoid null pointer exceptions.
```java
if ("rods".equals(name)) {
```

Reply:
Ahh makes sense, done.